### ISSUE TITLE
GroupID Not working in 6.3 the same as 6.2

### ISSUE THREAD

**Issue by fortoboy** [SCIInstitute/ShapeWorks#1781]
Hello, 

I am using a Mac, and two versions of shapeworks and getting different results
When I open a project file via .xlsx file in v6.2 I can properly definitely the groups by group_"namehere", however when I try to insert a group name in v6.3, shapeworks crashes when I select group set in analysis and when I export PCA component scores there is only 1 group identified although I have group_"namehere" identifying 2 groups of data. 

In summary, the problem is in v6.3 I cannot name the groups the same way I could in v6.2 by adding a group_"xx" tag to the excel spreadsheet without the program crashing upon trying to visualize the mean shape in each group.

Thanks,
Ron

**Comment by akenmorris** [comment]
Checking on it, there wasn't any intended change.

**Comment by akenmorris** [comment]
Is it possible that you could point me to a project that is crashing?  I've been trying to reproduce without luck so far.

Does the included example in Examples/Studio/FeatureMap/feature_map_example.xlsx also crash for you?

**Comment by fortoboy** [comment]
Hello, yes that example (Examples/Studio/FeatureMap/feature_map_example.xlsx) crashes on 6.3 but not 6.2 after running groom, optimize with all the default settings. Then when I go to the analyze tab, group set dropdown is available to press then when i select "side"

Here is the crash report generated by apple that contains my OS information:

Process:               ShapeWorksStudio [64867]
Path:                  /private/var/folders/*/ShapeWorksStudio.app/Contents/MacOS/ShapeWorksStudio
Identifier:            edu.utah.sci.shapeworks.studio
Version:                ()
Code Type:             X86-64 (Native)
Parent Process:        launchd [1]
User ID:               503

Date/Time:             2022-06-23 11:33:31.3101 -0400
OS Version:            macOS 12.4 (21
<img width="1217" alt="Screen Shot 2022-06-23 at 11 40 09 AM" src="https://user-images.githubusercontent.com/5419916/175339476-a5fa96d2-3ed6-428d-97fd-525c6c44e56b.png">
F79)
Report Version:        12
Bridge OS Version:     3.0 (14Y910)

Time Awake Since Boot: 30000 seconds
Time Since Wake:       8389 seconds
Crashed Thread:        2  QThread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
abort() called

**Comment by akenmorris** [comment]
Ah ok.  I am guessing this is from the new LDA chart creation that runs using Python.  If it can't initialize Python, it's supposed to show an error message with some instructions, but it looks like it's crashing.  Have you run the `source install_shapeworks.sh` script after install?

**Comment by fortoboy** [comment]
No I hadn’t, I directly opened the application after step 1 of the installation instructions. Are those steps 2 &3 necessary if only using the gui? Maybe useful to include in the instructions.

Install the ShapeWorks .pkg file by double-clicking on the file.

Open a terminal and change directory to the installation path.
cd /Applications/ShapeWorks

Create a protected conda environment that installs everything necessary to run.
source install_shapeworks.sh
Note: You can pass a different name for the environment, enabling multiple installations.
source install_shapeworks.sh shapeworks_61


Thank you,
Ronald Fortunato

**Comment by akenmorris** [comment]
It didn't used to be necessary, but some newer functionality is using Python packages for statistical analysis.

I will investigate a fix the crash and we'll create a patch to the 6.3 release.

**Comment by fortoboy** [comment]
An update, I ran the install_shapeworks.sh script as per the install
instructions to install a Conda environment called shapeworks_63 but that
did not fix the problem and I don’t know how the GUI would look for that
env.

**Comment by akenmorris** [comment]
The `install_shapeworks.sh` script is supposed to write a file `$HOME/.shapeworks/python_path.txt` that Studio looks for and uses with it's embedded interpreter to load python modules.  On windows it would be in %APPDATA% I believe.

If you load that conda env, can you run:

`python`
`import shapeworks`

**Comment by fortoboy** [comment]
Just to address your comment from last week, after loading the conda env I can run python and import shapeworks commands. Also I can confirm that $HOME/.shapeworks/python_path.txt was created and points to the env's library.

**Comment by akenmorris** [comment]
This should be fixed with #1797 and will be included in the upcoming 6.3.1 patch.

**Comment by akenmorris** [comment]
Fixed in 6.3.1
